### ISSUE TITLE
Translation Error For Fieldless Records Defined and Used in Separate Files

### ISSUE THREAD

**Issue by manthonyaiello** [lgwagner/SpeAR#42]
When a `FieldlessRecordExpr` appears in one file and is then used in a second file, `TranslateExpr` raises an exception, citing "Unexpected element provided."

The issue arises in `ReplaceShortHandRecords.transform`, where `EcoreUtil2.getAllContentsOfType(o, FieldlessRecordExpr.class);` returns an empty list: the search appears to only consider the current file and does not look in imported files. 

The attached files (which must be renamed to .spear) can be used to reproduce the error.
[fieldless-decl.txt](https://github.com/lgwagner/SpeAR/files/516617/fieldless-decl.txt)
[fieldless-use.txt](https://github.com/lgwagner/SpeAR/files/516618/fieldless-use.txt)

The exception trace is:

```
java.lang.RuntimeException: Unexpected element provided: com.rockwellcollins.spear.impl.FieldlessRecordExprImpl@1ab28416
    at com.rockwellcollins.spear.translate.lustre.TranslateExpr.defaultCase(TranslateExpr.java:260)
    at com.rockwellcollins.spear.translate.lustre.TranslateExpr.defaultCase(TranslateExpr.java:1)
    at com.rockwellcollins.spear.util.SpearSwitch.doSwitch(SpearSwitch.java:473)
    at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:53)
    at org.eclipse.emf.ecore.util.Switch.doSwitch(Switch.java:69)
    at com.rockwellcollins.spear.translate.lustre.TranslateExpr.translate(TranslateExpr.java:44)
    at com.rockwellcollins.spear.translate.master.SConstant.toLustre(SConstant.java:47)
    at com.rockwellcollins.spear.translate.master.SConstant.toLustre(SConstant.java:30)
    at com.rockwellcollins.spear.translate.master.SProgram.getBaseProgram(SProgram.java:125)
    at com.rockwellcollins.spear.translate.master.SProgram.getLogicalEntailment(SProgram.java:132)
    at com.rockwellcollins.spear.translate.actions.CheckLogicalEntailment$1.exec(CheckLogicalEntailment.java:102)
    at com.rockwellcollins.spear.translate.actions.CheckLogicalEntailment$1.exec(CheckLogicalEntailment.java:1)
    at org.eclipse.xtext.resource.OutdatedStateManager.exec(OutdatedStateManager.java:98)
    at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.internalReadOnly(XtextDocument.java:520)
    at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.readOnly(XtextDocument.java:492)
    at org.eclipse.xtext.ui.editor.model.XtextDocument.readOnly(XtextDocument.java:133)
    at com.rockwellcollins.spear.translate.actions.CheckLogicalEntailment.run(CheckLogicalEntailment.java:70)
    at org.eclipse.ui.internal.PluginAction.runWithEvent(PluginAction.java:247)
    at org.eclipse.ui.internal.WWinPluginAction.runWithEvent(WWinPluginAction.java:228)
    at org.eclipse.jface.action.ActionContributionItem.handleWidgetSelection(ActionContributionItem.java:565)
    at org.eclipse.jface.action.ActionContributionItem.lambda$5(ActionContributionItem.java:436)
    at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
    at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4248)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1501)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1524)
    at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1509)
    at org.eclipse.swt.widgets.Widget.notifyListeners(Widget.java:1313)
    at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4072)
    at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3698)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1121)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
    at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1022)
    at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:150)
    at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:687)
    at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
    at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:604)
    at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
    at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:138)
    at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
    at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
    at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
    at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
    at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
    at org.eclipse.equinox.launcher.Main.main(Main.java:1492)
```

**Comment by manthonyaiello** [comment]
Workaround: don't use fieldless record expressions. Instead of:

```
the_record is a record_type = new record_type {0, 1}
```

use

```
the_record is a record_type = new record_type {x = 0, y = 1}
```

**Comment by manthonyaiello** [comment]
Note that this is related to #41, but appears to be distinct.

**Comment by lgwagner** [comment]
fixed with the commit for #41
